src/main/java/org/alfresco/util/UrlUtil.java

```diff
@@ -70,6 +70,34 @@ public static String getShareUrl(SysAdminParams sysAdminParams)
                 sysAdminParams.getShareContext());
     }
 
+
+    public static String getApiExplorerUrl(SysAdminParams sysAdminParams)
+    {
+        return getApiExplorerUrl(sysAdminParams, "", "");
+    }
+
+    /**
+     * Builds URL to Api-Explorer based on the request only if the URL property is not provided
+     *  {@link SysAdminParams}.
+     * @return Rest-Api Url such as https://col.ab.or.ate/api-explorer/
+     *  or http://localhost:8082/api-explorer/
+     */
+    public static String getApiExplorerUrl(SysAdminParams sysAdminParams, String requestURL, String requestURI)
+    {
+        if (!sysAdminParams.getApiExplorerUrl().isEmpty())
+        {
+            return sysAdminParams.getApiExplorerUrl();
+        }
+        else
+        {
+            if (!requestURI.isEmpty() && !requestURL.isEmpty())
+            {
+                return requestURL.replace(requestURI,"/api-explorer");
+            }
+        }
+        return new String();
```

Comment:
you can just use "". it may help the compiler with the immutability of the string objects and all